torch/_dynamo/utils.py

```diff
@@ -925,10 +867,7 @@ def record_compilation_metrics(compilation_metrics: CompilationMetrics):
         name = "bwd_compilation_metrics"
     torch._logging.trace_structured(
         name,
-        lambda: {
-            k: list(v) if isinstance(v, set) else v
-            for k, v in dataclasses.asdict(compilation_metrics).items()
-        },
+        lambda: {k: list(v) if isinstance(v, set) else v for k, v in metrics.items()},
```

Comment:
huh what happened here?

Reply:
There's no need to convert the dict to a CompilationMetrics object and then convert it back to a dict with dataclasses.asdict when we already have the dict form. So that's all I changed (but I can also put it back)